dive_core/generateAdrenoHeader.py

```diff
@@ -257,19 +263,22 @@ def outputRegUnions(pm4_info_file, a6xx_domain, name, reg, for_pm4, postfix):
           
           pm4_info_file.write(extra_front_tab_str + "\tstruct\n") 
           pm4_info_file.write(extra_front_tab_str + "\t{\n") 
-          outputBitfields(pm4_info_file, bitset, extra_front_tab_str)
+          outputBitfields(pm4_info_file, bitset, extra_front_tab_str, shr_vars)
                   
   bitfields = reg.findall('{http://nouveau.freedesktop.org/}bitfield')
   # This handles the case where bitfields is used to extend bitset type, see SP_VS_CTRL_REG0
   if bitfields:
     if not use_bitset_as_type:
       pm4_info_file.write(extra_front_tab_str + "\tstruct\n") 
       pm4_info_file.write(extra_front_tab_str + "\t{\n")
-    outputBitfields(pm4_info_file, bitfields, extra_front_tab_str)
+    outputBitfields(pm4_info_file, bitfields, extra_front_tab_str, shr_vars)
 
   if use_bitset_as_type or bitfields:
     pm4_info_file.write(extra_front_tab_str + "\t}bitfields" + postfix + ";\n\n")
 
+  for index, shr_var in enumerate(shr_vars):
+    pm4_info_file.write(extra_front_tab_str + "\tenum {" + shr_var[0] + " = " + shr_var[1] + "};\n") 
```

Comment:
using enum so that i can define const inside union

Reply:
Yay! Another one of those things on my list that you're knocking out!

Can you show a before and after of the script results (i.e. code snippet from adreno.h) that shows what your change did?

Reply:
sure
![image](https://github.com/google/dive/assets/62302956/933d6622-edc7-47e6-a844-ed7d17c21741)

Reply:
ok, gcc doesn't like having enum declarations within the anonymous union, so i have to move the enum declarations outside, here is with the latest change:
![image](https://github.com/google/dive/assets/62302956/04202cc2-7704-4266-9d6b-837978ec909b)

Reply:
Thanks.
Is there a reason this isn't done as part of the script that generates pm4_info.h? That's usually where things like register "type" and other attributes are stored. It seems to be weird to put it into adreno.h instead.

Reply:
Specifically, I mean adding the field to this struct:

![image](https://github.com/google/dive/assets/42758786/4932247d-98fe-4418-92ff-df50c02e00cc)

Reply:
good point, i didn't know that we can put shift here. it is a much cleaner solution. will make the change, thanks!

Reply:
updated the change as you suggested

Reply:
Nitpicky: Do you mind cleaning up the history of this change?
You have a commit [Add support of the attribute of "shr"] which is reverted back soon after. It'd be better if that first commit didn't exist at all, if possible. Just to keep things clean.

Reply:
with squash, the history of those 2 commits would be gone?

Reply:
Oh that's true, if you're planning to squash *and* the latter commit 100% reverts back all the previous redundant code, then the squash would effectively eliminate that.

You can always squash manually on the git side to see if that's the case as well.